lib/modules/manager/gradle/artifacts.ts

```diff
@@ -153,17 +189,27 @@ export async function updateArtifacts({
       !updatedDeps.length ||
       isGcvPropsFile(packageFileName)
     ) {
-      cmd += ' --write-locks';
-    } else {
+      lockfileCmd += ' --write-locks';
+    } else if (lockFiles.length) {
       const updatedDepNames = updatedDeps
         .map(({ depName, packageName }) => packageName ?? depName)
         .filter(is.nonEmptyStringAndNotWhitespace);
 
-      cmd += ` --update-locks ${updatedDepNames.map(quote).join(',')}`;
+      lockfileCmd += ` --update-locks ${updatedDepNames.map(quote).join(',')}`;
     }
 
     await writeLocalFile(packageFileName, newPackageFileContent);
-    await exec(cmd, { ...execOptions, ignoreStdout: true });
+
+    const cmds = [lockfileCmd];
```

Comment:
Although it makes sense to be used together, gradle dependency verification also works when gradle lockfiles are not used. In that case, parsing the list of subprojects and calling gradle to update a lockfile are obsolete. Running the commands anyway does no harm besides prolonging renovate runs.

I think it would make sense to run lockfile-related commands only when lockfiles are present. Could be a follow-up refactoring PR but if someone else thinks this should be done here, that's fine for me too.

Reply:
sounds good